src/other/index.jl

```diff
@@ -108,19 +108,49 @@ function Base.insert!(x::Index, idx::Integer, nm::Symbol)
     x
 end
 
-Base.getindex(x::Index, idx::Symbol) = x.lookup[idx]
-Base.getindex(x::AbstractIndex, idx::Real) = Int(idx)
-Base.getindex(x::AbstractIndex, idx::AbstractVector{Union{Bool, Missing}}) =
-    getindex(x, collect(Missings.replace(idx, false)))
+Base.getindex(x::AbstractIndex, idx::Symbol) = x.lookup[idx]
+Base.getindex(x::AbstractIndex, idx::AbstractVector{Symbol}) = [x.lookup[i] for i in idx]
+
+function Base.getindex(x::AbstractIndex, idx::Real)
+    idx isa Bool && Base.depwarn("Indexing with Bool values is deprecated", :getindex)
```

Comment:
Better implement these deprecations via specialized methods in deprecated.jl, so that the code here doesn't need to be changed too much to remove deprecations.

Also, "Indexing with Bool values is deprecated" could be misleading: should specify "except for `Array{Bool}`".

Reply:
OK. The message is taken from Base but I will add the additional comment.